modules/openapi-generator/src/main/resources/dart/api_client.mustache

```diff
@@ -44,11 +44,7 @@ class ApiClient {
           {{#model}}
         case '{{classname}}':
             {{#isEnum}}
-          // Enclose the value in a list so that Dartson can use a transformer
-          // to decode it.
-          final listValue = [value];
-          final List<dynamic> listResult = dson.map(listValue, []);
-          return listResult[0];
+              return new {{classname}}TypeTransformer().decode(value);
```

Comment:
Suggestion: adjust the indention to make it the same as https://github.com/OpenAPITools/openapi-generator/pull/3048/files#diff-ff80004e8250c1792931642e1bf6d752R50